### ISSUE TITLE
Save state before sleep

### ISSUE THREAD

**Issue by regisfrias** [leinelissen/jellyfin-audio-player#125]
At least on iOS, after the app goes to sleep (automatically done by the OS if inactive for some time) it doesn't remember the previous state. So, for example, if I was listening to an album in the morning, in the evening I have to remember by myself where I left off. It would be good if the app would remember at least the album and the song that was being played before it went to sleep.

**Comment by StayBlue** [comment]
I think this could be in the form of a prompt when you open the app saying whether or not you want to load what you were previously listening to or not (I think a cool name would be like "Playback States."

**Comment by leinelissen** [comment]
Hey, thanks for the input! There is a sort of straightforward way to do this. However, there are some quirks, particularly that the app can be closed after being playing in the background for a considerable time. This make things somewhat harder, but let me think on it. If any of you feel like getting involved in development, I can give you some pointers where to start.

**Comment by regisfrias** [comment]
At some point I tried to run the project on my machine, but got errors. These were probably due to my M1 processor. I tried the most common fixes from the internet, but couldn't get it to work. I tried with other existing projects as well, with the same result. I suspect this might be related to the React Native version, since I'm able to create a new project.

If you have experience with that I'm all ears 🙂 I can't promise anything yet, but getting the project to run is a great first step. 

P.S.: I have no experience with React Native, only with ReactJS.

**Comment by leinelissen** [comment]
Yeah, that shouldn't be a problem. All versions of this app are built on an M1 Mac. Unfortunately, the React Native toolchain is really finicky to set up. Often the error messages you get don't really correspond to the problem at hand. I've updated all dependencies a couple months ago, and I have managed to get a local setup working with Node 16 and Xcode 14.2. If you have any concrete errors, let me know so I can help you debug them.

**Comment by regisfrias** [comment]
Actually, now that you mention, I didn't try other versions of Node itself. I'll try that later. But if you want you can already give me some directions for this particular issue (play state) and I can try to check it as soon as I have something working. (Again, not promising anything 😄 )

**Comment by regisfrias** [comment]
@leinelissen maybe you're more capable of deciphering this:

```
error Failed to build iOS project. We ran "xcodebuild" command but it exited with error code 65. To debug build logs further, consider building your app with Xcode.app, by opening Fintunes.xcworkspace.
Command line invocation:
    /Applications/Xcode.app/Contents/Developer/usr/bin/xcodebuild -workspace Fintunes.xcworkspace -configuration Debug -scheme Fintunes -destination id=CCA4C2D1-43A6-49FF-BEA1-0EDABBFE79D3

User defaults from command line:
    IDEPackageSupportUseBuiltinSCM = YES

Prepare packages

Computing target dependency graph and provisioning inputs

Create build description
Build description signature: f1e2fe67d6d3fb8720eef31935069caf
Build description path: /Users/regisfrias/Library/Developer/Xcode/DerivedData/Fintunes-dztgoiioimfnyxfsajujcpwmfiby/Build/Intermediates.noindex/XCBuildData/f1e2fe67d6d3fb8720eef31935069caf-desc.xcbuild

/Users/regisfrias/Projetos/jellyfin-audio-player/ios/Pods/Target Support Files/Pods-Fintunes/Pods-Fintunes.debug.xcconfig:1:1: error: unable to open configuration settings file
/Users/regisfrias/Projetos/jellyfin-audio-player/ios/Pods/Target Support Files/Pods-Fintunes/Pods-Fintunes.debug.xcconfig:1:1: error: unable to open configuration settings file
/Users/regisfrias/Projetos/jellyfin-audio-player/ios/Pods/Target Support Files/Pods-Fintunes/Pods-Fintunes.debug.xcconfig:1:1: error: unable to open configuration settings file
/Users/regisfrias/Projetos/jellyfin-audio-player/ios/Pods/Target Support Files/Pods-Fintunes/Pods-Fintunes.debug.xcconfig:1:1: error: unable to open configuration settings file
note: Building targets in dependency order
warning: Run script build phase 'Start Packager' will be run during every build because it does not specify any outputs. To address this warning, either add output dependencies to the script phase, or configure it to run in every build by unchecking "Based on dependency analysis" in the script phase. (in target 'Fintunes' from project 'Fintunes')

** BUILD FAILED **
```

I don't know if it's related, but the folder `jellyfin-audio-player/ios/Pods` doesn't exist. Whoever is looking for a config file there seems to be looking in the wrong place.

**Comment by leinelissen** [comment]
Ah this makes sense. In order for React Native to be able to deeply integrate native capabilities, it uses a bunch of native iOS dependencies. In the iOS world, those dependencies are usually managed with a thing called CocoaPods. Think of CocoaPods as NPM for Swift and Obj-C. In fact, for React Native this is the only way of installing native dependencies. What you'll need to do is install CocoaPods, for instance [via Homebrew](https://formulae.brew.sh/formula/cocoapods) and then run `pod install` in the `./ios` directory. This will install and setup all native dependencies, and that should resolve this particular error.

**Comment by regisfrias** [comment]
Yeah, actually I also had done that and got another error (sorry, forgot to mention that one):

```

[!] Invalid `Podfile` file:
[!] Invalid `RNGestureHandler.podspec` file: undefined method `exists?' for File:Class.

 #  from /Users/regisfrias/Projetos/jellyfin-audio-player/node_modules/react-native-gesture-handler/RNGestureHandler.podspec:5
 #  -------------------------------------------
 #
 >  isUserApp = File.exists?(File.join(__dir__, "..", "..", "node_modules", "react-native", "package.json"))
 #  if isUserApp
 #  -------------------------------------------
.

 #  from /Users/regisfrias/Projetos/jellyfin-audio-player/ios/Podfile:8
 #  -------------------------------------------
 #  target 'Fintunes' do
 >    config = use_native_modules!
 #
 #  -------------------------------------------
```

Any ideas?

**Comment by leinelissen** [comment]
This appears to be an issue with the latest Ruby version that is automatically installed with Homebrew CocoaPods: https://dev.to/retyui/fix-a-pod-install-error-undefined-method-exist-for-fileclass-react-native-24ke
